news/models.py

```diff
@@ -177,17 +185,54 @@ def copy_relations(self, oldinstance):
         """
         self.tags.set(oldinstance.tags.all())
 
-    def get_latest_articles(self, limit=3):
+    @property
+    def category_articles(self):
+        """
+        Return a queryset for a category
+        """
+        return (
+            Article.objects.published()
+            .filter(category__in=self.category)
+            .distinct()
+            .order_by("-created_at")[: self.num_articles]
```

Comment:
You could remove `[:self.num_articles]` from here and other methods and instead add it on to the return in your `get_articles` method so you only have it once `queryset[:self.num_articles]`